### PR TITLE
test suite works again on my mac

### PR DESCRIPTION
**Pull request by doutriaux1**
pep8 had broken all string formatting of errors

@durack1 please review

### CODE REVIEW

**Comment by durack1** [comment]
@doutriaux1 looks like `OS X` is having problems - `libcdms`?:
```
[ 47%] Performing configure step for 'libcdms'

CMake Error at /Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/libcdms-prefix/src/libcdms-stamp/libcdms-configure.cmake:56 (message):

  CTEST_FULL_OUTPUT

   Command failed: 1

   '/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/bin/cmake' '-DADDITIONAL_CFLAGS=-fPIC' '-DCONFIGURE_ARGS=--srcdir=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/libcdms;--enable-dap=;--enable-drs=no;--enable-hdf=no;--enable-pp=yes;--enable-ql=no;--cache-file=/dev/null;--prefix=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals;--with-nclib=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/lib;--with-ncinc=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/include;--with-daplib=/lib;--with-dapinc=/include;--with-hdfinc=./include;--with-hdflib=./lib;--with-hdf5lib=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/lib;--with-grib2lib=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/lib;--with-jasperlib=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/lib;--with-grib2inc=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/include;--enable-grib2' '-DINSTALL_DIR=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Externals' '-DWORKING_DIR=/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/libcdms' '-P' '/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/CMake/cdat_configure_step.cmake'

   From /Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/libcdms-prefix/src/libcdms-stamp/libcdms-configure-out.log
```
AND

```
make: *** [all] Error 2

 ERROR: Could not compile (make) cdat code

~/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat ~/build/PCMDI/_build/PCMDI_METRICS/tmp/metrics

Sorry...

This action did not complete successfully

Please try to re-run this task until successful before continuing further

 ERROR: Could not compile (make) cdat code

~/build/PCMDI/_build/PCMDI_METRICS/tmp/metrics

Sorry...

This action did not complete successfully

Please try to re-run this task until successful before continuing further
```

**Comment by durack1** [comment]
@doutriaux1 how about including `XQuartz` to solve the libcdms problem:

https://gist.github.com/BanzaiMan/5243f87d164ebb939eed

**Comment by durack1** [comment]
@doutriaux1 I've managed to get this over the `libcdms` hump by including `XQuartz` in the pre-install steps, but we've now got a failure with `ESMF`

**Comment by doutriaux1** [comment]
in the mean time we made sure you don't need it anymore for libcdms :wink:

**Comment by durack1** [comment]
@doutriaux1 I've tried to get this back up and running, however will have to divert back to you as I'm now stumped with next steps (and determining where it's falling over)..

There's also a bug in `installer_funcs.bash` as it's not picking up the existing `cmake` install as being valid - even though I've tweaked `install_metrics.bash` to be up to date with `3.3.2` as a max version:

```
Checking for CMake >=  2.8.11 
DEBUG: check_version_with $1='cmake'
min_version = [2.8.11]
max_version = [3.3.2]
DEBUG: check_version_helper $1='3.3.2' $2='2.8.11' $3='3.3.2'
DEBUG: check_version_between $1='3.3.2' $2='2.8.11' $3='3.3.2'
The return value from the call to helper function check_version_helper is 1
The detected version of cmake [3.3.2] is not between [2.8.11] and [3.3.2] 
*******************************
Setting up CMake 3.2.3
*******************************
```

**Comment by durack1** [comment]
This [travis tweak](http://docs.travis-ci.com/user/migrating-from-legacy/) could speed up testing..

**Comment by durack1** [comment]
@doutriaux1 seems the `expat` issue is still there:
```
CMake Error at /Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/setuptools-prefix/src/setuptools-stamp/setuptools-build.cmake:56 (message):
  CTEST_FULL_OUTPUT
   Command failed: 1
   'env' 'PYTHONPATH=' 'LD_LIBRARY_PATH=' '/Users/travis/build/PCMDI/_build/PCMDI_METRICS/bin/python' 'setup.py' 'build'
   From /Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/setuptools-prefix/src/setuptools-stamp/setuptools-build-out.log
  -------------------------------------------------
  -------------------------------------------------
   From /Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/setuptools-prefix/src/setuptools-stamp/setuptools-build-err.log
  -------------------------------------------------
  Traceback (most recent call last):
    File "setup.py", line 17, in <module>
      exec(init_file.read(), command_ns)
    File "<string>", line 8, in <module>
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/setuptools/__init__.py", line 11, in <module>
      from setuptools.extension import Extension
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/setuptools/extension.py", line 5, in <module>
      from setuptools.dist import _get_unpatched
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/setuptools/dist.py", line 16, in <module>
      import pkg_resources
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/pkg_resources.py", line 706, in <module>
      class Environment(object):
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/pkg_resources.py", line 709, in Environment
      def __init__(self, search_path=None, platform=get_supported_platform(), python=PY_MAJOR):
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/pkg_resources.py", line 148, in get_supported_platform
      plat = 'macosx-%s-%s' % ('.'.join(_macosx_vers()[:2]), m.group(3))
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/build/setuptools/pkg_resources.py", line 241, in _macosx_vers
      version = platform.mac_ver()[0]
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/platform.py", line 809, in mac_ver
      info = _mac_ver_xml()
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/platform.py", line 786, in _mac_ver_xml
      pl = plistlib.readPlist(fn)
    File "/Us-- [download 47% complete]
-- [download 58% complete]
-- [download 48% complete]
-- [download 49% complete]
-- [download 59% complete]
ers/travis/build/PCMDI/_build/PCMDI_METRICS/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plistlib.py", line 78, in readPlist
      rootObject = p.parse(pathOrFile)
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plistlib.py", line 401, in parse
      from xml.parsers.expat import ParserCreate
    File "/Users/travis/build/PCMDI/_build/PCMDI_METRICS/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/parsers/expat.py", line 4, in <module>
      from pyexpat import *
  ImportError: No module named pyexpat
```

**Comment by doutriaux1** [comment]
@durack1 seems like travis changed something the uv-cdat builds suddenly started to fail as well.

**Comment by durack1** [comment]
@doutriaux1 this PR needs a cleanup, the `json` files that correct the test case sign errors are fine, so to are the tweaks to `pcmdi_metrics_driver.py` but not the `travis.yml` and `install_metrics.bash` changes - can you remove these and I'll merge ASAP..

**Comment by doutriaux1** [comment]
@durack1 I did not touch the travis.yml file, I'm not sure how there got here. I'll copy the one from master bck in here, same for install_metrics.sh.

**Comment by durack1** [comment]
@doutriaux1 perfect, can you also review/merge https://github.com/UV-CDAT/uvcdat/pull/1674 so we can get that green `travis` badge back on?

**Comment by durack1** [comment]
@doutriaux1 as it's a metrics day it'd be good to get through as many of these cleanups as possible.. @gleckler1 is probably primed for a metrics lunch too..

@doutriaux1 let me know when you've updated the PR and I'll merge..

**Comment by durack1** [comment]
@doutriaux1 so https://github.com/UV-CDAT/uvcdat/pull/1674 is merged in and this should be our first green `travis` badge in a while (hopefully).. I'll await these tests and then merge..

**Comment by durack1** [comment]
@doutriaux1 there's still an [issue on linux](https://travis-ci.org/PCMDI/pcmdi_metrics/jobs/90830834):
```
CMAKE ARGS: cmake /home/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat -DCDAT_BUILD_MODE=LEAN -DCDAT_BUILD_ESMF_ESMP=ON -DCMAKE_INSTALL_PREFIX=/home/travis/build/PCMDI/_build/PCMDI_METRICS -DZLIB_PATCH_SRC=3 -DGIT_PROTOCOL=git://
PATH:/home/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/bin:/home/travis/bin:/home/travis/.local/bin:/home/travis/.gimme/versions/go1.4.1.linux.amd64/bin:/usr/local/phantomjs/bin:/home/travis/.nvm/v0.10.36/bin:./node_modules/.bin:/usr/local/maven-3.2.5/bin:/usr/local/clang-3.4/bin:/home/travis/.gimme/versions/go1.4.1.linux.amd64/bin:/usr/local/phantomjs/bin:./node_modules/.bin:/usr/local/maven-3.2.5/bin:/usr/local/clang-3.4/bin:/home/travis/.gimme/versions/go1.4.1.linux.amd64/bin:/home/travis/.rvm/gems/ruby-1.9.3-p551/bin:/home/travis/.rvm/gems/ruby-1.9.3-p551@global/bin:/home/travis/.rvm/rubies/ruby-1.9.3-p551/bin:/usr/local/phantomjs/bin:./node_modules/.bin:/usr/local/maven-3.2.5/bin:/usr/local/clang-3.4/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/travis/.rvm/bin:/home/travis/.rvm/bin:/home/travis/.rvm/bin
PWD:/home/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build
CMake Error: Could not find CMAKE_ROOT !!!
CMake has most likely not been installed correctly.
Modules directory not found in
/home/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/bin
CMake Error: Error executing cmake::LoadCache(). Aborting.
 ERROR: Could not configure (cmake) cdat code (2)
~/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat ~/build/PCMDI/_build/PCMDI_METRICS/tmp/metrics
Sorry...
This action did not complete successfully
Please try to re-run this task until successful before continuing further
```

**Comment by durack1** [comment]
@doutriaux1 you're getting there, the `travis` `OS X` build seemed to sail through but still has [two test errors](https://travis-ci.org/PCMDI/pcmdi_metrics/jobs/90830854#L9323)

**Comment by doutriaux1** [comment]
@durack1 now all tests pass on my ubuntu and my mac. Let's wait and see what travis says

**Comment by durack1** [comment]
@doutriaux1 great - fingers crossed!

**Comment by durack1** [comment]
@doutriaux1 no good? [Same `cmake` error on linux for both builds](https://travis-ci.org/PCMDI/pcmdi_metrics/jobs/90854522#L4135-L4140)
```
PWD:/home/travis/build/PCMDI/_build/PCMDI_METRICS/tmp/uvcdat/uvcdat_build
CMake Error: Could not find CMAKE_ROOT !!!
CMake has most likely not been installed correctly.
Modules directory not found in
/home/travis/build/PCMDI/_build/PCMDI_METRICS/Externals/bin
CMake Error: Error executing cmake::LoadCache(). Aborting.
```

File: test/pcmdi/installationTest/tas_2.5x2.5_regrid2_linear_metrics.json [hunk 1]
```diff
@@ -389,6 +389,6 @@
             "Name": "Custom Maximum"
         }
     },
-    "metrics_git_sha1": "40e056c9ec4b9ec96eb5c3a93ad89fcd427429f5",
-    "uvcdat_version": "2.1.0-682-g5f296b7"
+    "metrics_git_sha1": "3100dca207554bffd8cc9c56b7d3887c47a3f14b",
+    "uvcdat_version": "uvcdat-2.2.0-320-g80914b7"
```
Comment (durack1) [inline test/pcmdi/installationTest/tas_2.5x2.5_regrid2_linear_metrics.json]: @doutriaux1 is this ok - being built with a non-updated version of the metrics package? I wonder if the numbers might change subtly the other `jsons` are created using `2.4.0.rc2-56-g2a38096`

Comment (doutriaux1) [inline test/pcmdi/installationTest/tas_2.5x2.5_regrid2_linear_metrics.json]: the test checker ignores these bits.

Comment (durack1) [inline test/pcmdi/installationTest/tas_2.5x2.5_regrid2_linear_metrics.json]: My question was more about the numbers in the `json` file, as we know roundoff and masking can cause these to move around a bit.. `2.2.0` vs `2.4.0` might give subtly different numbers that ARE being tested..

**Comment by durack1** [comment]
@doutriaux1 any luck getting `cmake` working again on `travis` linux?

**Comment by doutriaux1** [comment]
not until after the release

**Comment by durack1** [comment]
@doutriaux1 I'm reluctant to merge until the travis builds are passing fine.. Do you know what is going on here?

**Comment by doutriaux1** [comment]
@durack1 it is a separate issue, you can reluctantly let the PR accumulate in the mean time :wink: I would suggest opening a separate issue for this cmake thing though since it has nothing to do with the metrics code itself, just the install.

**Pull request merged**